site/docs/v1/tech/email-templates/templates-replacement-variables.adoc

```diff
@@ -700,6 +700,9 @@ ${code}
 [.api]
 [field]#application# [type]#[Application]# [since]#Available since 1.28.0#::
 The Application object, see the link:/docs/v1/tech/apis/applications[Application API] for field definitions.
++
+*Note*:
+This object may not be available depending upon when this template is constructed. If you utilize this object in your template, ensure you first check to see if it is defined. You can check for this variable safely in freemarker by wrapping the variable as such: `${(application)!""}`.
```

Comment:
I read this as a user and say "how would I know know if the application is available or not?" Is there an easy way to tell them?

Reply:
Yes, we should be able to tell the user how they can ensure this reference is available.

Reply:
Notes for myself. `application` is not available when:
- Using the `POST /api/two-factor/start` API without supplying `applicationId` on the request
- Using `POST /api/login` without supplying `applicationId` (doc points out there is probably no production use case for this)
- Enabling or disabling the email MFA method (it uses the same template)